jina/hubble/hubio.py

```diff
@@ -588,11 +587,23 @@ def fetch_meta(
             payload['tag'] = tag
 
         req_header = get_request_header()
-        resp = requests.post(pull_url, json=payload, headers=req_header)
-        if resp.status_code != 200:
-            if resp.text:
-                raise Exception(resp.text)
-            resp.raise_for_status()
+
+        logger = JinaLogger('HubIO.fetch_meta')
+        for i in range(3):
+            try:
+                resp = requests.post(pull_url, json=payload, headers=req_header)
+                if resp.status_code != 200:
+                    if resp.text:
+                        raise Exception(resp.text)
+                    resp.raise_for_status()
```

Comment:
can we wrap the retry as a function, and can reuse this function in some other places.

Reply:
Created `retry` decorator in `helpers`.

The exact failure/retry login in `fetch_meta` doesn't apply to other `HubIO` methods.
So I created a new function inside `fetch_meta`.